snap/info.go

```diff
@@ -397,6 +397,13 @@ type PlugInfo struct {
 	Hooks     map[string]*HookInfo
 }
 
+func (plug *PlugInfo) Attr(key string) (interface{}, error) {
```

Comment:
I guess the caller will do a type assert because this is returning `interface{}`. In such case you may refactor this into
```
import "reflect"
...
func (plug *PlugInfo) Attr(key string, val interface{}) error {
	rt := reflect.TypeOf(val)
	if rt.Kind() != reflect.Ptr || val == nil {
		return errors.New("not assignable")
	}
	if v, ok := plug.Attrs[name]; ok {
		if reflect.TypeOf(v) != rt.Elem() {
			return errors.New("type mismatch")
		}
		rv := reflect.ValueOf(val)
		rv.Elem().Set(reflect.ValueOf(v))
		return nil
	}
	return errors.New("not found")
}
```

Then the call sites will be cleaner:

```
var foo string
if err := plug.Attr("foo", &foo); err != nil {
    // blah blah failed
}
```

A working example is right here: https://play.golang.org/p/tyvQ0RpeFA

Reply:
Thanks, I like that, it would simplify the code of interfaces a little bit. @pedronis , @niemeyer your opinion on that? If that's ok I'll update the ConnectedPlug/ConnectedSlot to match.